src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditProceedingsAndCostsSectionController.java

```diff
@@ -25,63 +61,679 @@
+   */
+  private void populateFurtherDetailsDropdowns(
+      final Model model,
+      final ApplicationDetail application,
+      final ProceedingFormDataProceedingDetails proceedingDetails,
+      final String matterType) {
+
+    final String proceedingCode = proceedingDetails.getProceedingType();
+
+    final Mono<List<ClientInvolvementTypeLookupValueDetail>> clientInvolvementTypesMono =
+        lookupService.getClientInvolvementTypes(proceedingCode)
+            .map(ClientInvolvementTypeLookupDetail::getContent);
+
+    final Mono<List<LevelOfServiceLookupValueDetail>> levelOfServiceTypesMono =
+        lookupService.getProceedingLevelOfServiceTypes(application, proceedingCode, matterType)
+            .map(LevelOfServiceLookupDetail::getContent);
+
+    Mono.zip(clientInvolvementTypesMono, levelOfServiceTypesMono)
+        .doOnNext(tuple -> {
+          model.addAttribute("clientInvolvementTypes", tuple.getT1());
+          model.addAttribute("levelOfServiceTypes", tuple.getT2());
+        }).block();
+
+    if (proceedingDetails.getOrderTypeRequired() != null
+        && proceedingDetails.getOrderTypeRequired()) {
+      new DropdownBuilder(model)
+          .addDropdown("orderTypes",
+              lookupService.getOrderTypes())
+          .build();
+    }
+
+  }
+
+  /**
+   * Handles the GET request to confirm the action (add or edit) on a proceeding.
+   *
+   * @param application The application details.
+   * @param action The action being performed (add or edit).
+   * @param model The Model object to add attributes to for the view.
+   * @param session The HTTP session.
+   * @return The name of the view to be rendered.
+   */
+  @GetMapping("/application/summary/proceedings/{action}/confirm")
+  public String proceedingsActionConfirm(
+      @SessionAttribute(APPLICATION) final ApplicationDetail application,
+      @PathVariable("action") final String action,
+      final Model model,
+      final HttpSession session) {
+
+    ProceedingFlowFormData proceedingFlow = null;
+
+    //coming directly from proceeding summary - editing scope limitations
+    if (isRoutedFromSummaryPage(action, session)) {
+      final Proceeding proceeding = (Proceeding) session.getAttribute(CURRENT_PROCEEDING);
+
+      final String orderTypeDisplayValue = Optional.ofNullable(proceeding.getTypeOfOrder())
+          .map(StringDisplayValue::getId)
+          .map(id -> lookupService.getOrderTypeDescription(id).block())
+          .orElse(null);
+
+      proceedingFlow = proceedingAndCostsMapper.toProceedingFlow(proceeding, orderTypeDisplayValue);
+
+      proceedingFlow.setEditingScopeLimitations(true);
+
+      model.addAttribute(PROCEEDING_FLOW_FORM_DATA, proceedingFlow);
+      model.addAttribute(CURRENT_PROCEEDING, proceeding);
+
+    } else {
+      //Came from previous step in the flow
+      proceedingFlow = (ProceedingFlowFormData) session.getAttribute(PROCEEDING_FLOW_FORM_DATA);
+    }
+
+    if (proceedingFlow.isAmended()) {
```

Comment:
This section could do with a few comments. I'm struggling to work out what it's doing.

Reply:
Added comments